torch/csrc/utils/tensor_numpy.cpp

```diff
@@ -129,8 +129,17 @@ at::Tensor tensor_from_numpy(PyObject* obj) {
   if (!PyArray_Check(obj)) {
     throw TypeError("expected np.ndarray (got %s)", Py_TYPE(obj)->tp_name);
   }
-
   auto array = (PyArrayObject*)obj;
+
+  if (!PyArray_ISWRITEABLE(array)) {
+    TORCH_WARN(
```

Comment:
perhaps we should TORCH_WARN_ONCE?  I'm not really sure I can think of a case where we wouldn't want to warn once...users really don't like getting spammed.

Reply:
Updated.